pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,232 @@
+        validator=truncated_range,
+        values=[-3.5, 3.5]
+    )
+
+    current_limit = Instrument.control(
+        "SOUR:CURR:LIM?", "SOUR:CURR:LIM %g",
+        """ Sets an upper soft limit on the programmed output current
+        for the supply. Can take values floating point from 0 and +3.5 A. """,
+        validator=truncated_range,
+        values=[0, 3.5]
+    )    
+
+
+
+    ###############
+    # Voltage (V) #
+    ###############
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        """ Returns the actual voltage in Volts.
+        """
+    )
+    voltage_setpoint = Instrument.control(
+        "SOUR:VOLT?", "SOUR:VOLT %g",
+        """ A floating point property that immediately sets the desired voltage output 
+        in Volts, which can take floating point values between 300 and 300 V. """,
+        validator=truncated_range,
+        values=[-300, 300]
+    )
+
+    voltage_limit = Instrument.control(
+        "SOUR:VOLT:LIM?", "SOUR:VOLT:LIM %g",
+        """ Sets an upper soft limit on the programmed output voltage
+        for the supply. Can take values floating point from 0 and 300 V. """,
+        validator=truncated_range,
+        values=[0, 300]
+    )    
+
+
+    def __init__(self, adapter, **kwargs):
+        super(Sorenson30035E, self).__init__(
+            adapter, "Sorenson 300-3.5E DC Power Supply", **kwargs
+        )
+
+    def ramp_to_voltage(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+         if abs(value) > 300:
+            raise ValueError("Requested voltage too large, |V| must be 300 V or less")
+         self.write("SOUR:VOLT:RAMP %g %g" % (value, time))
+
+    def ramp_to_current(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+         if abs(value) > 3.5:
+            raise ValueError("Requested current too large, |I| must be 3.5 A or less")
+         self.write("SOUR:CURR:RAMP %g %g" % (value, time))
+
+    def abort_ramp(self):
+        """ Aborts all in-progress ramps
+        """
+        self.write("SOUR:CURR:RAMP:ABOR")
+
+    def enable_source(self):
+        """ Enables the output of the power supply (switch isolation relays on) """
+        if self.source_enable == False:
+            self.source_enable = True
+        else:
+            pass
+
+    def disable_source(self):
+        """ Disables the  output in an unprotected way. DO NOT use this carelessly, use shutdown,
+        it's safer."""
```

Comment:
If this is unsafe, it's probably better to not implement this, especially if a preferable alternative is available.